// fakergem/lib/faker.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var address_1 = require("./faker/address");
var ancient_1 = require("./faker/ancient");
var app_1 = require("./faker/app");
var avatar_1 = require("./faker/avatar");
var bank_1 = require("./faker/bank");
var beer_1 = require("./faker/beer");
var bitcoin_1 = require("./faker/bitcoin");
var book_1 = require("./faker/book");
var boolean_1 = require("./faker/boolean");
var business_1 = require("./faker/business");
var cat_1 = require("./faker/cat");
var chuck_norris_1 = require("./faker/chuck-norris");
var coffee_1 = require("./faker/coffee");
var color_1 = require("./faker/color");
var commerce_1 = require("./faker/commerce");
var companies_house_1 = require("./faker/companies-house");
var company_1 = require("./faker/company");
var compass_1 = require("./faker/compass");
var crypto_1 = require("./faker/crypto");
var date_1 = require("./faker/date");
var demographic_1 = require("./faker/demographic");
var dragon_ball_1 = require("./faker/dragon-ball");
var educator_1 = require("./faker/educator");
var fake_1 = require("./faker/fake");
var file_1 = require("./faker/file");
var fillmurray_1 = require("./faker/fillmurray");
var food_1 = require("./faker/food");
var friends_1 = require("./faker/friends");
var game_of_thrones_1 = require("./faker/game-of-thrones");
var hacker_1 = require("./faker/hacker");
var harry_potter_1 = require("./faker/harry-potter");
var hipster_1 = require("./faker/hipster");
var internet_1 = require("./faker/internet");
var lord_of_the_rings_1 = require("./faker/lord-of-the-rings");
var lorem_1 = require("./faker/lorem");
var lorem_flickr_1 = require("./faker/lorem-flickr");
var lorem_pixel_1 = require("./faker/lorem-pixel");
var matz_1 = require("./faker/matz");
var music_1 = require("./faker/music");
var name_1 = require("./faker/name");
var number_1 = require("./faker/number");
var phone_number_1 = require("./faker/phone-number");
var placeholdit_1 = require("./faker/placeholdit");
var pokemon_1 = require("./faker/pokemon");
var random_1 = require("./faker/random");
var rick_and_morty_1 = require("./faker/rick-and-morty");
var rock_band_1 = require("./faker/rock-band");
var ru_paul_1 = require("./faker/ru-paul");
var slack_emoji_1 = require("./faker/slack-emoji");
var space_1 = require("./faker/space");
var star_wars_1 = require("./faker/star-wars");
var superhero_1 = require("./faker/superhero");
var team_1 = require("./faker/team");
var time_1 = require("./faker/time");
var twin_peaks_1 = require("./faker/twin-peaks");
var twitter_1 = require("./faker/twitter");
var university_1 = require("./faker/university");
var vehicle_1 = require("./faker/vehicle");
var zelda_1 = require("./faker/zelda");
var Faker = /** @class */ (function () {
    function Faker() {
        this.Address = new address_1.Address(this);
        this.Ancient = new ancient_1.Ancient(this);
        this.App = new app_1.App(this);
        this.Avatar = new avatar_1.Avatar(this);
        this.Bank = new bank_1.Bank(this);
        this.Beer = new beer_1.Beer(this);
        this.Bitcoin = new bitcoin_1.Bitcoin(this);
        this.Book = new book_1.Book(this);
        this.Boolean = new boolean_1.Boolean(this);
        this.Business = new business_1.Business(this);
        this.Cat = new cat_1.Cat(this);
        this.ChuckNorris = new chuck_norris_1.ChuckNorris(this);
        this.Coffee = new coffee_1.Coffee(this);
        this.Color = new color_1.Color(this);
        this.Commerce = new commerce_1.Commerce(this);
        this.CompaniesHouse = new companies_house_1.CompaniesHouse(this);
        this.Company = new company_1.Company(this);
        this.Compass = new compass_1.Compass(this);
        this.Crypto = new crypto_1.Crypto(this);
        this.Date = new date_1.DateFaker(this);
        this.Demographic = new demographic_1.Demographic(this);
        this.DragonBall = new dragon_ball_1.DragonBall(this);
        this.Educator = new educator_1.Educator(this);
        this.Fake = new fake_1.Fake(this);
        this.File = new file_1.File(this);
        this.Fillmurray = new fillmurray_1.Fillmurray(this);
        this.Food = new food_1.Food(this);
        this.Friends = new friends_1.Friends(this);
        this.GameOfThrones = new game_of_thrones_1.GameOfThrones(this);
        this.Hacker = new hacker_1.Hacker(this);
        this.HarryPotter = new harry_potter_1.HarryPotter(this);
        this.Hipster = new hipster_1.Hipster(this);
        this.Internet = new internet_1.Internet(this);
        this.LordOfTheRings = new lord_of_the_rings_1.LordOfTheRings(this);
        this.LoremFlickr = new lorem_flickr_1.LoremFlickr(this);
        this.LoremPixel = new lorem_pixel_1.LoremPixel(this);
        this.Lorem = new lorem_1.Lorem(this);
        this.Matz = new matz_1.Matz(this);
        this.Music = new music_1.Music(this);
        this.Name = new name_1.Name(this);
        this.Number = new number_1.Number(this);
        this.PhoneNumber = new phone_number_1.PhoneNumber(this);
        this.Placeholdit = new placeholdit_1.Placeholdit(this);
        this.Pokemon = new pokemon_1.Pokemon(this);
        this.Random = new random_1.Random(this);
        this.RickAndMorty = new rick_and_morty_1.RickAndMorty(this);
        this.RockBand = new rock_band_1.RockBand(this);
        this.RuPaul = new ru_paul_1.RuPaul(this);
        this.SlackEmoji = new slack_emoji_1.SlackEmoji(this);
        this.Space = new space_1.Space(this);
        this.StarWars = new star_wars_1.StarWars(this);
        this.Superhero = new superhero_1.Superhero(this);
        this.Team = new team_1.Team(this);
        this.Time = new time_1.Time(this);
        this.TwinPeaks = new twin_peaks_1.TwinPeaks(this);
        this.Twitter = new twitter_1.Twitter(this);
        this.University = new university_1.University(this);
        this.Vehicle = new vehicle_1.Vehicle(this);
        this.Zelda = new zelda_1.Zelda(this);
    }
    return Faker;
}());
exports.Faker = Faker;
