// fakergem/lib/faker/avatar.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var Avatar = /** @class */ (function () {
    function Avatar(faker) {
        this.SUPPORTED_FORMATS = ['png', 'jpg', 'bmp'];
        this.faker = faker;
    }
    Avatar.prototype.image = function (slug, size, format, set, bgset) {
        if (slug === void 0) { slug = null; }
        if (size === void 0) { size = '300x300'; }
        if (format === void 0) { format = 'png'; }
        if (set === void 0) { set = 'set1'; }
        if (bgset === void 0) { bgset = null; }
        if (!this.isValidSize(size)) {
            throw new Error('Size should be specified in format 300x300');
        }
        if (!this.isFormatSupported(format)) {
            throw new Error("Supported formats are " + this.SUPPORTED_FORMATS.join(', '));
        }
        var filename = (slug || this.words()) + "." + format;
        var bgsetQuery = bgset ? "&bgset=" + bgset : '';
        return "https://robohash.org/" + filename + "?size=" + size + "&set=" + set + bgsetQuery;
    };
    Avatar.prototype.isValidSize = function (size) {
        return /^\d+x\d+$/.test(size);
    };
    Avatar.prototype.isFormatSupported = function (format) {
        return this.SUPPORTED_FORMATS.indexOf(format) >= 0;
    };
    Avatar.prototype.words = function () {
        return this.faker.Lorem.words(3).join('-');
    };
    return Avatar;
}());
exports.Avatar = Avatar;
