// fakergem/lib/faker/beer.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var beer_json_1 = __importDefault(require("../data/beer.json"));
var Beer = /** @class */ (function () {
    function Beer(faker) {
        this.faker = faker;
    }
    Beer.prototype.name = function () {
        return this.faker.Random.element(beer_json_1.default.names);
    };
    Beer.prototype.style = function () {
        return this.faker.Random.element(beer_json_1.default.styles);
    };
    Beer.prototype.hop = function () {
        return this.faker.Random.element(beer_json_1.default.hops);
    };
    Beer.prototype.yeast = function () {
        return this.faker.Random.element(beer_json_1.default.yeasts);
    };
    Beer.prototype.malt = function () {
        return this.faker.Random.element(beer_json_1.default.malts);
    };
    Beer.prototype.ibu = function () {
        return this.faker.Number.between(10, 99) + " IBU";
    };
    Beer.prototype.alcohol = function () {
        return this.faker.Number.betweenF(2.0, 10.0).toFixed(1) + "%";
    };
    Beer.prototype.blg = function () {
        return this.faker.Number.betweenF(2.0, 10.0).toFixed(1) + "\u00B0Blg";
    };
    return Beer;
}());
exports.Beer = Beer;
