// fakergem/lib/faker/bitcoin.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var bs58_1 = __importDefault(require("bs58"));
var crypto_1 = __importDefault(require("crypto"));
var Bitcoin = /** @class */ (function () {
    function Bitcoin(faker) {
        this.PROTOCOL_VERSION = {
            main: 0,
            testnet: 111,
        };
        this.faker = faker;
    }
    Bitcoin.prototype.address = function () {
        return this.addressFor(this.PROTOCOL_VERSION.main);
    };
    Bitcoin.prototype.testnetAddress = function () {
        return this.addressFor(this.PROTOCOL_VERSION.testnet);
    };
    Bitcoin.prototype.addressFor = function (version) {
        var packed = (String.fromCharCode(version) + crypto_1.default.randomBytes(20));
        var packedDigest = this.digest(packed);
        var checksum = this.digest(packedDigest).slice(0, 3);
        return this.base58("" + packed + checksum);
    };
    Bitcoin.prototype.digest = function (text) {
        var hash = crypto_1.default.createHash('sha256');
        return hash.digest(text);
    };
    Bitcoin.prototype.base58 = function (text) {
        return bs58_1.default.encode(new Buffer(text, 'ascii'));
    };
    return Bitcoin;
}());
exports.Bitcoin = Bitcoin;
