// fakergem/lib/faker/business.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var business_json_1 = __importDefault(require("../data/business.json"));
var Business = /** @class */ (function () {
    function Business(faker) {
        this.faker = faker;
    }
    Business.prototype.creditCardNumber = function () {
        return this.faker.Random.element(business_json_1.default.creditCardNumbers);
    };
    Business.prototype.creditCardExpiryDate = function () {
        return this.daysFromNow(365 * this.faker.Number.between(1, 5));
    };
    Business.prototype.creditCardType = function () {
        return this.faker.Random.element(business_json_1.default.creditCardTypes);
    };
    Business.prototype.daysFromNow = function (n) {
        var d = new Date();
        d.setDate(d.getDate() + n);
        d.setHours(0, 0, 0, 0);
        return d;
    };
    return Business;
}());
exports.Business = Business;
