// fakergem/lib/faker/coffee.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var coffee_json_1 = __importDefault(require("../data/coffee.json"));
var Coffee = /** @class */ (function () {
    function Coffee(faker) {
        this.faker = faker;
    }
    Coffee.prototype.blendName = function () {
        var blendName = this.faker.Random.element(coffee_json_1.default.blendNames);
        return this.parse(blendName);
    };
    Coffee.prototype.origin = function () {
        var country = this.faker.Random.element(coffee_json_1.default.countries);
        var key = country.replace(/\s/, '_').toLowerCase();
        var regions = coffee_json_1.default.regions;
        var region = this.faker.Random.element(regions[key]);
        return region + ", " + country;
    };
    Coffee.prototype.variety = function () {
        return this.faker.Random.element(coffee_json_1.default.varieties);
    };
    Coffee.prototype.notes = function () {
        var notes = this.faker.Random.element(coffee_json_1.default.notes);
        return this.parse(notes);
    };
    Coffee.prototype.parse = function (format) {
        return format
            .replace(/\{intensifier\}/g, this.faker.Random.element(coffee_json_1.default.intensifiers))
            .replace(/\{body\}/g, this.faker.Random.element(coffee_json_1.default.bodies))
            .replace(/\{descriptor\}/g, this.faker.Random.element(coffee_json_1.default.descriptors))
            .replace(/\{name1\}/g, this.faker.Random.element(coffee_json_1.default.name1))
            .replace(/\{name2\}/g, this.faker.Random.element(coffee_json_1.default.name2));
    };
    return Coffee;
}());
exports.Coffee = Coffee;
