// fakergem/lib/faker/color.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var color_json_1 = __importDefault(require("../data/color.json"));
var text_helpers_1 = require("../utils/text-helpers");
var Color = /** @class */ (function () {
    function Color(faker) {
        this.faker = faker;
    }
    Color.prototype.hexColor = function () {
        var n = this.faker.Number.between(0, 0xffffff);
        return "#" + text_helpers_1.leftPad(n.toString(16), 6, '0');
    };
    Color.prototype.colorName = function () {
        return this.faker.Random.element(color_json_1.default.colorNames);
    };
    Color.prototype.rgbColor = function () {
        var _this = this;
        return Array(3)
            .fill(null)
            .map(function (_) { return _this.faker.Number.between(0, 255); });
    };
    Color.prototype.hslColor = function () {
        return [
            this.faker.Number.between(0, 360),
            Math.round(this.faker.Number.betweenF(0.0, 1.0)),
            Math.round(this.faker.Number.betweenF(0.0, 1.0)),
        ];
    };
    Color.prototype.hslaColor = function () {
        return [
            this.faker.Number.between(0, 360),
            Math.round(this.faker.Number.betweenF(0.0, 1.0)),
            Math.round(this.faker.Number.betweenF(0.0, 1.0)),
            this.faker.Number.betweenF(0.0, 1.0),
        ];
    };
    return Color;
}());
exports.Color = Color;
