// fakergem/lib/faker/compass.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var compass_json_1 = __importDefault(require("../data/compass.json"));
var Compass = /** @class */ (function () {
    function Compass(faker) {
        this.faker = faker;
    }
    Compass.prototype.direction = function () {
        return this.parse(this.faker.Random.element(compass_json_1.default.directions));
    };
    Compass.prototype.cardinal = function () {
        return this.faker.Random.element(compass_json_1.default.cardinals.word);
    };
    Compass.prototype.ordinal = function () {
        return this.faker.Random.element(compass_json_1.default.ordinals.word);
    };
    Compass.prototype.halfWind = function () {
        return this.faker.Random.element(compass_json_1.default.halfWinds.word);
    };
    Compass.prototype.quarterWind = function () {
        return this.faker.Random.element(compass_json_1.default.quarterWinds.word);
    };
    Compass.prototype.abbreviation = function () {
        return this.parse(this.faker.Random.element(compass_json_1.default.abbreviations));
    };
    Compass.prototype.cardinalAbbreviation = function () {
        return this.faker.Random.element(compass_json_1.default.cardinals.abbreviation);
    };
    Compass.prototype.ordinalAbbreviation = function () {
        return this.faker.Random.element(compass_json_1.default.ordinals.abbreviation);
    };
    Compass.prototype.halfWindAbbreviation = function () {
        return this.faker.Random.element(compass_json_1.default.halfWinds.abbreviation);
    };
    Compass.prototype.quarterWindAbbreviation = function () {
        return this.faker.Random.element(compass_json_1.default.quarterWinds.abbreviation);
    };
    Compass.prototype.azimuth = function () {
        return this.parse(this.faker.Random.element(compass_json_1.default.azimuths));
    };
    Compass.prototype.cardinalAzimuth = function () {
        return this.faker.Random.element(compass_json_1.default.cardinals.azimuth);
    };
    Compass.prototype.ordinalAzimuth = function () {
        return this.faker.Random.element(compass_json_1.default.ordinals.azimuth);
    };
    Compass.prototype.halfWindAzimuth = function () {
        return this.faker.Random.element(compass_json_1.default.halfWinds.azimuth);
    };
    Compass.prototype.quarterWindAzimuth = function () {
        return this.faker.Random.element(compass_json_1.default.quarterWinds.azimuth);
    };
    Compass.prototype.parse = function (format) {
        var text = format.replace(/\{(\w+)\}/g, function (m) { return "{Compass." + m.substring(1); });
        return this.faker.Fake.f(text);
    };
    return Compass;
}());
exports.Compass = Compass;
