// fakergem/lib/faker/crypto.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var crypto_1 = __importDefault(require("crypto"));
var Crypto = /** @class */ (function () {
    function Crypto(faker) {
        this.CHARACTERS = '0123456789abcdefghijklmnopqrstuvwxyz';
        this.faker = faker;
    }
    Crypto.prototype.md5 = function () {
        var hash = crypto_1.default.createHash('md5');
        var array = this.CHARACTERS.split('');
        var characters = this.faker.Random.assortment(array, 255).join('');
        hash.update(characters);
        return hash.digest('hex');
    };
    Crypto.prototype.sha1 = function () {
        var hash = crypto_1.default.createHash('sha1');
        var array = this.CHARACTERS.split('');
        var characters = this.faker.Random.assortment(array, 255).join('');
        hash.update(characters);
        return hash.digest('hex');
    };
    Crypto.prototype.sha256 = function () {
        var hash = crypto_1.default.createHash('sha256');
        var array = this.CHARACTERS.split('');
        var characters = this.faker.Random.assortment(array, 255).join('');
        hash.update(characters);
        return hash.digest('hex');
    };
    return Crypto;
}());
exports.Crypto = Crypto;
