// fakergem/lib/faker/demographic.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var demographic_json_1 = __importDefault(require("../data/demographic.json"));
var Demographic = /** @class */ (function () {
    function Demographic(faker) {
        this.faker = faker;
    }
    Demographic.prototype.race = function () {
        return this.faker.Random.element(demographic_json_1.default.races);
    };
    Demographic.prototype.educationalAttainment = function () {
        return this.faker.Random.element(demographic_json_1.default.educationalAttainments);
    };
    Demographic.prototype.demonym = function () {
        return this.faker.Random.element(demographic_json_1.default.demonyms);
    };
    Demographic.prototype.maritalStatus = function () {
        return this.faker.Random.element(demographic_json_1.default.maritalStatuses);
    };
    Demographic.prototype.sex = function () {
        return this.faker.Random.element(demographic_json_1.default.sexes);
    };
    Demographic.prototype.height = function (unit) {
        if (unit === void 0) { unit = 'metric'; }
        switch (unit) {
            case 'metric':
                return "" + this.faker.Number.betweenF(1.45, 2.13).toFixed(2);
            case 'imperial':
                var inches = this.faker.Number.between(57, 86);
                return Math.floor(inches / 12) + " ft, " + inches % 12 + " in";
            default:
                return '0.00';
        }
    };
    return Demographic;
}());
exports.Demographic = Demographic;
