// fakergem/lib/faker/educator.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var educator_json_1 = __importDefault(require("../data/educator.json"));
var Educator = /** @class */ (function () {
    function Educator(faker) {
        this.faker = faker;
    }
    Educator.prototype.university = function () {
        var name = this.faker.Random.element(educator_json_1.default.names);
        var tertiaryType = this.faker.Random.element(educator_json_1.default.tertiaries.types);
        return name + " " + tertiaryType;
    };
    Educator.prototype.secondarySchool = function () {
        var name = this.faker.Random.element(educator_json_1.default.names);
        var secondary = this.faker.Random.element(educator_json_1.default.secondaries);
        return name + " " + secondary;
    };
    Educator.prototype.course = function () {
        var courseType = this.faker.Random.element(educator_json_1.default.tertiaries.course.types);
        var courseSubject = this.faker.Random.element(educator_json_1.default.tertiaries.course.subjects);
        return courseType + " " + courseSubject;
    };
    Educator.prototype.campus = function () {
        var name = this.faker.Random.element(educator_json_1.default.names);
        return name + " Campus";
    };
    return Educator;
}());
exports.Educator = Educator;
