// fakergem/lib/faker/game-of-thrones.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var game_of_thrones_json_1 = __importDefault(require("../data/game-of-thrones.json"));
var GameOfThrones = /** @class */ (function () {
    function GameOfThrones(faker) {
        this.faker = faker;
    }
    GameOfThrones.prototype.character = function () {
        return this.faker.Random.element(game_of_thrones_json_1.default.characters);
    };
    GameOfThrones.prototype.house = function () {
        return this.faker.Random.element(game_of_thrones_json_1.default.houses);
    };
    GameOfThrones.prototype.city = function () {
        return this.faker.Random.element(game_of_thrones_json_1.default.cities);
    };
    GameOfThrones.prototype.dragon = function () {
        return this.faker.Random.element(game_of_thrones_json_1.default.dragons);
    };
    return GameOfThrones;
}());
exports.GameOfThrones = GameOfThrones;
