// fakergem/lib/faker/hacker.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var hacker_json_1 = __importDefault(require("../data/hacker.json"));
var Hacker = /** @class */ (function () {
    function Hacker(faker) {
        this.faker = faker;
    }
    Hacker.prototype.saySomethingSmart = function () {
        return this.parse(this.faker.Random.element(hacker_json_1.default.phrases));
    };
    Hacker.prototype.abbreviation = function () {
        return this.faker.Random.element(hacker_json_1.default.abbreviations);
    };
    Hacker.prototype.adjective = function () {
        return this.faker.Random.element(hacker_json_1.default.adjectives);
    };
    Hacker.prototype.noun = function () {
        return this.faker.Random.element(hacker_json_1.default.nouns);
    };
    Hacker.prototype.verb = function () {
        return this.faker.Random.element(hacker_json_1.default.verbs);
    };
    Hacker.prototype.ingverb = function () {
        return this.faker.Random.element(hacker_json_1.default.ingverbs);
    };
    Hacker.prototype.parse = function (format) {
        var text = format.replace(/\{(\w+)\}/g, function (m) { return "{Hacker." + m.substring(1); });
        return this.faker.Fake.f(text).replace(/^(\w)/, function (m) { return m[0].toUpperCase(); });
    };
    return Hacker;
}());
exports.Hacker = Hacker;
