// fakergem/lib/faker/harry-potter.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var harry_potter_json_1 = __importDefault(require("../data/harry-potter.json"));
var HarryPotter = /** @class */ (function () {
    function HarryPotter(faker) {
        this.faker = faker;
    }
    HarryPotter.prototype.character = function () {
        return this.faker.Random.element(harry_potter_json_1.default.characters);
    };
    HarryPotter.prototype.location = function () {
        return this.faker.Random.element(harry_potter_json_1.default.locations);
    };
    HarryPotter.prototype.quote = function () {
        return this.faker.Random.element(harry_potter_json_1.default.quotes);
    };
    HarryPotter.prototype.book = function () {
        return this.faker.Random.element(harry_potter_json_1.default.books);
    };
    return HarryPotter;
}());
exports.HarryPotter = HarryPotter;
