// fakergem/lib/faker/lorem.js
"use strict";
var __spreadArrays = (this && this.__spreadArrays) || function () {
    for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
    for (var r = Array(s), k = 0, i = 0; i < il; i++)
        for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
            r[k] = a[j];
    return r;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var lorem_json_1 = __importDefault(require("../data/lorem.json"));
var text_helpers_1 = require("../utils/text-helpers");
// 0-9, a-z
var CHARACTERS = '0123456789abcdefghijklmnopqrstuvwxyz';
var Lorem = /** @class */ (function () {
    function Lorem(faker) {
        this.faker = faker;
    }
    Lorem.prototype.word = function () {
        return this.faker.Random.element(lorem_json_1.default.words);
    };
    Lorem.prototype.words = function (num, supplemental) {
        if (num === void 0) { num = 3; }
        if (supplemental === void 0) { supplemental = false; }
        var wordList = supplemental ? __spreadArrays(lorem_json_1.default.words, [lorem_json_1.default.supplemental]) : lorem_json_1.default.words;
        var words = this.faker.Random.assortment(wordList, Math.max(0, num));
        return words;
    };
    Lorem.prototype.character = function () {
        return this.faker.Random.element(CHARACTERS);
    };
    Lorem.prototype.characters = function (charCount) {
        var _this = this;
        if (charCount === void 0) { charCount = 255; }
        return Array(Math.max(0, charCount))
            .fill(null)
            .map(function () { return _this.character(); })
            .join('');
    };
    Lorem.prototype.sentence = function (wordCount, supplemental, randomWordsToAdd) {
        if (wordCount === void 0) { wordCount = 4; }
        if (supplemental === void 0) { supplemental = false; }
        if (randomWordsToAdd === void 0) { randomWordsToAdd = 6; }
        var text = text_helpers_1.capitalize(this.words(Math.max(0, wordCount) + this.faker.Number.between(0, Math.max(0, randomWordsToAdd)), supplemental).join(' '));
        return text.length > 0 ? text + "." : text;
    };
    Lorem.prototype.sentences = function (sentenceCount, supplemental) {
        var _this = this;
        if (sentenceCount === void 0) { sentenceCount = 3; }
        if (supplemental === void 0) { supplemental = false; }
        return Array(Math.max(0, sentenceCount))
            .fill(null)
            .map(function (_) { return _this.sentence(3, supplemental); });
    };
    Lorem.prototype.paragraph = function (sentenceCount, supplemental, randomSentencesToAdd) {
        if (sentenceCount === void 0) { sentenceCount = 3; }
        if (supplemental === void 0) { supplemental = false; }
        if (randomSentencesToAdd === void 0) { randomSentencesToAdd = 3; }
        return this.sentences(sentenceCount + this.faker.Number.between(0, randomSentencesToAdd), supplemental).join(' ');
    };
    Lorem.prototype.paragraphs = function (paragraphCount, supplemental) {
        var _this = this;
        if (paragraphCount === void 0) { paragraphCount = 3; }
        if (supplemental === void 0) { supplemental = false; }
        return Array(Math.max(0, paragraphCount))
            .fill(null)
            .map(function (_) { return _this.paragraph(3, supplemental); });
    };
    return Lorem;
}());
exports.Lorem = Lorem;
