// fakergem/lib/faker/lorem-pixel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var LoremPixel = /** @class */ (function () {
    function LoremPixel(faker) {
        this.SUPPORTED_CATEGORIES = [
            'abstract',
            'animals',
            'business',
            'cats',
            'city',
            'food',
            'nightlife',
            'fashion',
            'people',
            'nature',
            'sports',
            'technics',
            'transport',
        ];
        this.faker = faker;
    }
    LoremPixel.prototype.image = function (size, isGray, category, num, text) {
        if (size === void 0) { size = '300x300'; }
        if (isGray === void 0) { isGray = false; }
        if (category === void 0) { category = null; }
        if (num === void 0) { num = null; }
        if (text === void 0) { text = null; }
        if (!this.isValidSize(size)) {
            throw new Error('Size should be specified in format 300x300');
        }
        if (!this.isCategorySupported(category)) {
            throw new Error("Supported categories are " + this.SUPPORTED_CATEGORIES.join(', '));
        }
        if (!this.isNumberValid(num)) {
            throw new Error('Number must be between 1 and 10');
        }
        if (num && !category) {
            throw new Error('Category required when number is passed');
        }
        if (text && !category && !num) {
            throw new Error('Category and number must be passed when text is passed');
        }
        return [
            'http://lorempixel.com',
            isGray ? 'g' : null,
            size.replace(/x/, '/'),
            [category, num, text].filter(function (x) { return x; }).join('/'),
        ]
            .filter(function (x) { return x; })
            .join('/');
    };
    LoremPixel.prototype.isValidSize = function (size) {
        return /^\d+x\d+$/.test(size);
    };
    LoremPixel.prototype.isCategorySupported = function (category) {
        return !category || this.SUPPORTED_CATEGORIES.indexOf(category) >= 0;
    };
    LoremPixel.prototype.isNumberValid = function (num) {
        return (num === null ||
            Array(10)
                .fill(null)
                .map(function (_, i) { return i + 1; })
                .indexOf(num) >= 0);
    };
    return LoremPixel;
}());
exports.LoremPixel = LoremPixel;
