// fakergem/lib/faker/lorem-flickr.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var LoremFlickr = /** @class */ (function () {
    function LoremFlickr(faker) {
        this.SUPPORTED_COLORIZATIONS = ['red', 'green', 'blue'];
        this.faker = faker;
    }
    LoremFlickr.prototype.image = function (size, searchTerms, matchAll) {
        if (size === void 0) { size = '300x300'; }
        if (searchTerms === void 0) { searchTerms = []; }
        if (matchAll === void 0) { matchAll = false; }
        return this.buildUrl(size, null, searchTerms, matchAll);
    };
    LoremFlickr.prototype.grayscaleImage = function (size, searchTerms, matchAll) {
        if (size === void 0) { size = '300x300'; }
        if (searchTerms === void 0) { searchTerms = ['all']; }
        if (matchAll === void 0) { matchAll = false; }
        if (!this.hasSearchTerms(searchTerms)) {
            throw new Error('Search terms must be specified for grayscale images');
        }
        return this.buildUrl(size, 'g', searchTerms, matchAll);
    };
    LoremFlickr.prototype.pixelatedImage = function (size, searchTerms, matchAll) {
        if (size === void 0) { size = '300x300'; }
        if (searchTerms === void 0) { searchTerms = ['all']; }
        if (matchAll === void 0) { matchAll = false; }
        if (!this.hasSearchTerms(searchTerms)) {
            throw new Error('Search terms must be specified for pixelated images');
        }
        return this.buildUrl(size, 'p', searchTerms, matchAll);
    };
    LoremFlickr.prototype.colorizedImage = function (size, color, searchTerms, matchAll) {
        if (size === void 0) { size = '300x300'; }
        if (color === void 0) { color = 'red'; }
        if (searchTerms === void 0) { searchTerms = ['all']; }
        if (matchAll === void 0) { matchAll = false; }
        if (!this.hasSearchTerms(searchTerms)) {
            throw new Error('Search terms must be specified for colorized images');
        }
        if (!this.isColorizationSupported(color)) {
            throw new Error("Supported colorizations are " + this.SUPPORTED_COLORIZATIONS.join(', '));
        }
        return this.buildUrl(size, color, searchTerms, matchAll);
    };
    LoremFlickr.prototype.buildUrl = function (size, format, searchTerms, matchAll) {
        if (!this.isValidSize(size)) {
            throw new Error('Size should be specified in format 300x300');
        }
        var url = 'http://loremflickr.com';
        if (format) {
            url += "/" + format;
        }
        url += "/" + size.replace('x', '/');
        if ((searchTerms || []).length > 0) {
            url += "/" + searchTerms.join(',');
        }
        if (matchAll) {
            url += '/all';
        }
        return url;
    };
    LoremFlickr.prototype.hasSearchTerms = function (searchTerms) {
        return (searchTerms || []).length > 0;
    };
    LoremFlickr.prototype.isColorizationSupported = function (color) {
        return this.SUPPORTED_COLORIZATIONS.indexOf(color) >= 0;
    };
    LoremFlickr.prototype.isValidSize = function (size) {
        return /^\d+x\d+$/.test(size);
    };
    return LoremFlickr;
}());
exports.LoremFlickr = LoremFlickr;
