// fakergem/lib/faker/name.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var name_json_1 = __importDefault(require("../data/name.json"));
var Name = /** @class */ (function () {
    function Name(faker) {
        this.faker = faker;
    }
    Name.prototype.name = function () {
        return [this.firstName(), this.lastName()].join(' ');
    };
    Name.prototype.nameWithMiddle = function () {
        return [this.firstName(), this.firstName(), this.lastName()].join(' ');
    };
    Name.prototype.firstName = function () {
        return this.faker.Random.element(name_json_1.default.firstNames);
    };
    Name.prototype.lastName = function () {
        return this.faker.Random.element(name_json_1.default.lastNames);
    };
    Name.prototype.prefix = function () {
        return this.faker.Random.element(name_json_1.default.prefixes);
    };
    Name.prototype.suffix = function () {
        return this.faker.Random.element(name_json_1.default.suffixes);
    };
    Name.prototype.title = function () {
        return [
            this.faker.Random.element(name_json_1.default.titles.descriptor),
            this.faker.Random.element(name_json_1.default.titles.level),
            this.faker.Random.element(name_json_1.default.titles.job),
        ].join(' ');
    };
    return Name;
}());
exports.Name = Name;
