// fakergem/lib/faker/phone-number.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var phone_number_json_1 = __importDefault(require("../data/phone-number.json"));
var PhoneNumber = /** @class */ (function () {
    function PhoneNumber(faker) {
        this.faker = faker;
    }
    PhoneNumber.prototype.phoneNumber = function () {
        var _this = this;
        var format = this.faker.Random.element(phone_number_json_1.default.phoneNumber.formats);
        return format.replace(/#/g, function () { return _this.faker.Number.between(0, 9); });
    };
    PhoneNumber.prototype.cellPhone = function () {
        var _this = this;
        var format = this.faker.Random.element(phone_number_json_1.default.cellPhone.formats);
        return format.replace(/#/g, function () { return _this.faker.Number.between(0, 9); });
    };
    PhoneNumber.prototype.areaCode = function () {
        return this.faker.Random.element(phone_number_json_1.default.areaCodes);
    };
    PhoneNumber.prototype.exchangeCode = function () {
        return this.faker.Random.element(phone_number_json_1.default.exchangeCodes);
    };
    PhoneNumber.prototype.subscriberNumber = function (length) {
        if (length === void 0) { length = 4; }
        var min = Math.pow(10, length) / 10;
        var max = min * 10 - 1;
        return "" + this.faker.Number.between(min, max);
    };
    PhoneNumber.prototype.extension = function (length) {
        if (length === void 0) { length = 4; }
        return this.subscriberNumber(length);
    };
    return PhoneNumber;
}());
exports.PhoneNumber = PhoneNumber;
