// fakergem/lib/faker/space.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var space_json_1 = __importDefault(require("../data/space.json"));
var Space = /** @class */ (function () {
    function Space(faker) {
        this.faker = faker;
    }
    Space.prototype.planet = function () {
        return this.faker.Random.element(space_json_1.default.planets);
    };
    Space.prototype.moon = function () {
        return this.faker.Random.element(space_json_1.default.moons);
    };
    Space.prototype.galaxy = function () {
        return this.faker.Random.element(space_json_1.default.galaxies);
    };
    Space.prototype.nebula = function () {
        return this.faker.Random.element(space_json_1.default.nebulas);
    };
    Space.prototype.starCluster = function () {
        return this.faker.Random.element(space_json_1.default.starClusters);
    };
    Space.prototype.constellation = function () {
        return this.faker.Random.element(space_json_1.default.constellations);
    };
    Space.prototype.star = function () {
        return this.faker.Random.element(space_json_1.default.stars);
    };
    Space.prototype.agency = function () {
        return this.faker.Random.element(space_json_1.default.agencies);
    };
    Space.prototype.agencyAbv = function () {
        return this.faker.Random.element(space_json_1.default.agencyAbvs);
    };
    Space.prototype.nasaSpaceCraft = function () {
        return this.faker.Random.element(space_json_1.default.nasaSpaceCrafts);
    };
    Space.prototype.company = function () {
        return this.faker.Random.element(space_json_1.default.companies);
    };
    Space.prototype.distanceMeasurement = function () {
        return this.faker.Random.element(space_json_1.default.distanceMeasurements);
    };
    Space.prototype.meteorite = function () {
        return this.faker.Random.element(space_json_1.default.meteorites);
    };
    return Space;
}());
exports.Space = Space;
