// fakergem/lib/faker/star-wars.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var star_wars_json_1 = __importDefault(require("../data/star-wars.json"));
var text_helpers_1 = require("../utils/text-helpers");
var StarWars = /** @class */ (function () {
    function StarWars(faker) {
        this.faker = faker;
    }
    StarWars.prototype.character = function () {
        return this.faker.Random.element(star_wars_json_1.default.characters);
    };
    StarWars.prototype.droid = function () {
        return this.faker.Random.element(star_wars_json_1.default.droids);
    };
    StarWars.prototype.planet = function () {
        return this.faker.Random.element(star_wars_json_1.default.planets);
    };
    StarWars.prototype.quote = function () {
        return this.faker.Random.element(star_wars_json_1.default.quotes);
    };
    StarWars.prototype.specie = function () {
        return this.faker.Random.element(star_wars_json_1.default.species);
    };
    StarWars.prototype.vehicle = function () {
        return this.faker.Random.element(star_wars_json_1.default.vehicles);
    };
    StarWars.prototype.wookieSentence = function () {
        var _this = this;
        return text_helpers_1.capitalize(Array(this.faker.Number.between(1, 10))
            .fill(null)
            .map(function (_) { return _this.faker.Random.element(star_wars_json_1.default.wookieWords); })
            .join(' ') + this.faker.Random.element(['.', '?', '!']));
    };
    return StarWars;
}());
exports.StarWars = StarWars;
