// fakergem/lib/faker/superhero.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var superhero_json_1 = __importDefault(require("../data/superhero.json"));
var Superhero = /** @class */ (function () {
    function Superhero(faker) {
        this.faker = faker;
    }
    Superhero.prototype.name = function () {
        return this.faker.Random.element(superhero_json_1.default.names)
            .replace('{prefix}', this.faker.Random.element(superhero_json_1.default.prefixes))
            .replace('{suffix}', this.faker.Random.element(superhero_json_1.default.suffixes))
            .replace('{descriptor}', this.faker.Random.element(superhero_json_1.default.descriptors));
    };
    Superhero.prototype.power = function () {
        return this.faker.Random.element(superhero_json_1.default.powers);
    };
    Superhero.prototype.prefix = function () {
        return this.faker.Random.element(superhero_json_1.default.prefixes);
    };
    Superhero.prototype.suffix = function () {
        return this.faker.Random.element(superhero_json_1.default.suffixes);
    };
    Superhero.prototype.descriptor = function () {
        return this.faker.Random.element(superhero_json_1.default.descriptors);
    };
    return Superhero;
}());
exports.Superhero = Superhero;
