// fakergem/lib/faker/vehicle.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var vehicle_json_1 = __importDefault(require("../data/vehicle.json"));
var Vehicle = /** @class */ (function () {
    function Vehicle(faker) {
        this.VIN_CHARS = '0123456789.ABCDEFGH..JKLMN.P.R..STUVWXYZ';
        this.VIN_MAP = '0123456789X';
        this.VIN_WEIGHTS = '8765432X098765432';
        this.faker = faker;
    }
    Vehicle.prototype.vin = function () {
        var _this = this;
        var details = this.faker.Random.element(vehicle_json_1.default.manufactures.map(function (x) { return ({ wmi: x[1], wmiExt: x[2] }); }));
        var vin = Array(14)
            .fill(null)
            .map(function (_) { return _this.faker.Random.element(_this.VIN_CHARS.split('').filter(function (x) { return x !== '.'; })); })
            .join('');
        var buffer = ("" + details.wmi + vin).split('');
        if (details.wmiExt) {
            for (var i = 0; i < 2; i++) {
                buffer[i + 12] = details.wmiExt[i];
            }
        }
        buffer[10] = this.faker.Random.element(vehicle_json_1.default.years);
        buffer[8] = this.vinChecksum(buffer);
        return buffer.join('');
    };
    Vehicle.prototype.manufacture = function () {
        return this.faker.Random.element(vehicle_json_1.default.manufactures.map(function (x) { return x[0]; }));
    };
    Vehicle.prototype.vinChecksum = function (buffer) {
        var _this = this;
        return this.VIN_MAP[buffer.map(function (c, i) { return _this.calculateVinWeight(c, i); }) % 11];
    };
    Vehicle.prototype.calculateVinWeight = function (character, i) {
        return (this.VIN_CHARS.indexOf(character) % 10) * this.VIN_MAP.indexOf(this.VIN_WEIGHTS[i]);
    };
    return Vehicle;
}());
exports.Vehicle = Vehicle;
